tests/middleware/test_message_state.py

```diff
@@ -14,16 +16,30 @@ class TestMessageState:
     MessageState
     """
 
+    @freeze_time("1927-03-06", auto_tick_seconds=1)
     def test_pending_state_message(self, stub_broker, state_middleware, do_work):
         msg = do_work.send()
-        assert state_middleware.backend.get_state(msg.message_id).name == StateNamesEnum.Pending
+        state = state_middleware.backend.get_state(msg.message_id)
```

Comment:
I understand why you chose to use auto_tick_seconds. But it lie of the fact that datetime.now is called multiple times. So your are basically counting the number of time .now() is called. I would prefer that you changed the datetime manually between the states. Things would be more clear and adding a date.now later in the app would not change the tests.
I made a small exmaple
```
    def test_pending_state_message(self, stub_broker, state_middleware, do_work, frozen_datetime):
        msg = do_work.send()
        state = state_middleware.backend.get_state(msg.message_id)
        assert state.message_id == msg.message_id
        assert state.name == StateNamesEnum.Pending
        # do work takes one second from [0, 1]s then enqueue should be in second [1, 2]s
        assert state.enqueued_datetime.isoformat() == "2020-02-03T00:00:00+00:00"

  def test_success_state_message(self, stub_broker, stub_worker, state_middleware, do_work, frozen_datetime):
        msg = do_work.send()
        frozen_datetime.tick(delta=15)
        stub_broker.join(do_work.queue_name)
        stub_worker.join()
        state = state_middleware.backend.get_state(msg.message_id)
        assert state.name == StateNamesEnum.Success
        # do work takes 1 second, then the state be enqueed from [1, 2]s
        assert state.enqueued_datetime.isoformat() == "2020-02-03T00:00:00+00:00"
        # three operations, start the actor, save the actor and finish actor
        # then that should take 3 seconds
        assert state.end_datetime.isoformat() == "2020-02-03T00:00:15+00:00"
```
You can see that i use isofomat to test for the dates as i find if more lissible.
And in confest.py
```
@pytest.fixture
def frozen_datetime():
    with freeze_time("2020-02-03") as frozen_datetime:
        yield frozen_datetime
```
We could a a parameter to the fixture but it may not be very usefull for now.